### DISCUSSION TITLE
Anyone using this with doobie?

### DISCUSSION THREAD

**Discussion by rparree** [Iltotore/iron#185] [General]
I am migrating a project which has a lot of doobie repos. Did anyone get iron to work with it?

I am getting 

```
Found:    (e.toto : model.Toto)
Required: doobie.syntax.SqlInterpolator.SingleFragment[?] [69:224]
```

Tried to add the following implicits:

```
given [A,T: IronType[A, *]](using ev: Put[A]): Put[T]= ev.asInstanceOf[Put[T]]
given [A,T: IronType[A, *]](using ev: Get[A]): Get[T]= ev.asInstanceOf[Get[T]]
given [A,T: IronType[A, *]](using ev: Meta[A]): Meta[T]= ev.asInstanceOf[Meta[T]]
```

But i must be missing something (perhaps obvious, it has been a long week).

**Comment by Iltotore** [comment]
I am not familiar with Doobie but where does your `*` type comes from? If `T` is supposed to be the type to "serialize" then I think your givens should look like this:

```scala
given [A, C](using ev: Put[A]): Put[IronType[A, C]]= ev.asInstanceOf
```

**Comment by rparree** [comment]
I don't know what i was thinking. I copied similar lines of code from the previous version where we had are own "New Type" approach that used a bunch of typeclasses.  As i said, it was a long week :)

Thanks
